### ISSUE TITLE
Spring Boot Integration test failing

### ISSUE THREAD

**Issue by kunjalshah** [lukas-krecan/ShedLock#240]
1. Which version do you use - 4.0.0
2. Which Lock Provider - mysql JDBCTemplate
3. ShedLock configuration -  lockAtLeastFor = "14M", lockAtMostFor = "14M"

This is not actually a bug or an issue but I wanted to get some ideas on a failing integration test on my spring boot application. The integration test is using SpringJunit4ClassRunner and @WebIntegrationTest to do a end-end integration test. 

I am trying to add @SchedulerLock on an existing @Scheduled spring method and the existing integration test we have is failing after adding the shedlock annotation. I wanted to get some thoughts on best practices on writing a unit/integration test for shedlock framework. Is there a way to ignore the @ShedulerLock annotation in a spring integration test?

**Comment by lukas-krecan** [comment]
Can you please provide stack trace or an error message of the failure

**Comment by kunjalshah** [comment]
Thank you for the quick reply. There is no error message or stack trace in the logs per se other than my integration test fails with Assertion when I have the @SchedulerLock annotation on the scheduled method...  I do see the below trace in the console log when the integration test is run so not sure if the error/stack trace is masked/hidden?....
```
log4j:WARN No appenders could be found for logger (net.javacrumbs.shedlock.core.DefaultLockingTaskExecutor).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```
 I have the DEBUG trace set for log4j.logger.net.javacrumbs.shedlock=DEBUG in the test log4j.properties file. My integration test is pretty simple it uses awaility API to wait for a minute for the background @Scheduled task to finish and verify some data in the db the assertion failes ... 

see below code

> @Test
>     public void testWorkerThreadsUpdateEndpointSubTypeInformationForAll() throws InterruptedException {
> //        LockAssert.TestHelper.makeAllAssertsPass(true);
>         //Verify that there are no endpoints with stale subtype info after all the worker threads are finished (1 min timeout)
>         await().atMost(Duration.ONE_MINUTE)
>                 .untilAsserted(() -> assertTrue(endpointService.fetchEndpointsWithStaleSubTypeInfo(new PageRequest(0,workerManager.batchSize), workerManager.noOfDays).isEmpty()));
>     }

**Comment by lukas-krecan** [comment]
Well, it seems that the task is actually locked. There are two options how to fix it. 
1. Set  lockAtLeastFor and  lockAtMostFor to some lower value for tests using spring properties (both values support SpEL)
2. Use mock lockProvider in tests

**Comment by kunjalshah** [comment]
Okay, I think I got the log4j configured properly. It looks like the lock is obtained and released by checking the console log when integration test is run. But I do see this error in my integration test when the @SchedulerLock Annotation is added. I am not really sure why it says EntityManagerFactory is being destroyed? 

> 2020-05-04-pool-4-thread-1--DEBUG-net.javacrumbs.shedlock.core.DefaultLockingTaskExecutor:Task finished, lock 'EndpointSubTypeUpdateWorkerManager_startWorkerThreads' will be released at 2020-05-04T18:13:42.785Z
> 2020-05-04-pool-4-thread-1--ERROR-org.springframework.scheduling.support.TaskUtils$LoggingErrorHandler:Unexpected error occurred in scheduled task.
> org.springframework.beans.factory.BeanCreationNotAllowedException: Error creating bean with name 'entityManagerFactory': Singleton bean creation not allowed while the singletons of this factory are in destruction (Do not request a bean from a BeanFactory in a destroy method implementation!)
>
